client/src/components/PacmanGame/PacmanGame.jsx

```diff
@@ -24,35 +28,87 @@ class PacmanGame extends Component {
       for (const [x, y] of entityLocations) {
         gridState[x][y] = entityCode;
       }
-      // return gridState;
     };
 
-
-    // apply entities duummy data
     entityApplier(getFoods(), entityToCode('food'));
     entityApplier(getPacman(), entityToCode('pacman'));
-    console.log((gridState[43][5]));
-    // console.log(gridState[43])
     entityApplier(getWalls(cellsInEachRow), entityToCode('wall'));
     entityApplier(getGhosts(), entityToCode('ghost'));
-    console.log((gridState[43][5]));
     entityApplier(getEnergizers(), entityToCode('energizer'));
 
-    this.setState({ 'gridState': gridState });
-    this.setState({ 'gridSize': gridSize });
+    this.setState({ gridState, gridSize });
 
-    console.log(getPacman());
-    console.log((gridState[43][5]));
-    console.log('food', (gridState[15][9]));
+    let walls = [];
+    gridState.forEach((rowArray, rowIndex) => {
+      rowArray.forEach((cell, columnIndex) => {
+        if (cell === 4) {
+          walls.push([rowIndex, columnIndex]);
+        }
+      })
+    });
+    this.setState({ walls });
+  }
+  startGame = () => {
+    this.animationHandler = setInterval(
+      this.animateGame,
+      this.state.config.refreshRate
+    );
+    document.addEventListener('keydown', this.setDirection);
+  }
+  checkCollision = ({ x, y }) => {
+    let walls = this.state.walls;
+    let hasCollision = false;
 
-    // console.log('food', getFoods());
-    console.log('pacman', getPacman());
-    // console.log(getWalls());
-    console.log('ghost', getGhosts());
+    walls.forEach((wall) => {
```

Comment:
No need to iterate on all the wall cells to check for collision. Just check whether `codeToEntity( gridState[x][y] ) === 'wall'`. This will lead to only a constant time operation.